lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -268,9 +268,26 @@ LogicalResult TfheRustBoolEmitter::printSksMethod(
     }
 
     emitAssignPrefix(result);
+    os << variableNames->getNameForValue(sks);
 
-    os << variableNames->getNameForValue(sks) << ".packed_gates(\n";
-    os << "&vec![";
+    // parse the not gate
+    if (!gateStr.compare("NOT")) {
+      os << ".packed_not(\n";
+
+      os << commaSeparatedValues({nonSksOperands[0]}, [&](Value value) {
+        std::string prefix;
+        std::string suffix;
+
+        tie(prefix, suffix) = checkOrigin(value);
```

Comment:
TIL that `tie` is a thing!

Reply:
Yeah, it was the only way of producing a tuple that works with MLIR and TableGen